features/FxA-98-improve-fraud-detection/README.md

```diff
@@ -0,0 +1,122 @@
+- Feature Name: Improve fraud detection
+- Start Date: 2016-05-25
+- Feature PR:
+- Feature Milestone:
+
+# Summary
+[summary]: #summary
+
+# Motivation
+[motivation]: #motivation
+
+We wish to increase security for FxA by improving the fraud detection the customs server can provide. We want a /v2 API that is more logically structured, and take the opportunity to do the same to the internal server code.
+
+We wish to utilize a more permanent data store so we can better detech fraud.
+
+The customs server v2 will lend itself to be more easily configurable and manageable (ie, Adding custom rules, manually block). Ops would like to be able to adapt during a prolonged attack, and have configuration update without rebuilding the server.
+
+## What Do We Gain
+
+- **detect access from new IP addresses**: The customs server will save IP address/email combinations, so that it can detect when a successful login attempt is from a "new" IP. The auth-server confirmation emails can be reduced to only send to users if login occurs from a "new" IP address.
+- **incorporate 3rd party block lists**: Hand-wavey statistics suggest that 66% of fraudulent login attempts come from IP addresses that are already on these block lists.
+
+# Detailed design
+[design]: #detailed-design
+
+## Database
+[database]: #db
+
+V1 uses memcached, which means an attacker can try to push results out of our memory db. This may be sufficient for short term fraud detection, such as count number of login attempts for a certain IP in the past few hours.
+
+However, we wish to add a persistent data store, such as MySQL, to save data for longer periods. The use case in this feature is to save all unique IP addresses used with a particular email address. This allows us to implement monitoring of when an unknown or new IP address tries to do something with a particular email.
+
+### Tables
+
+- `grants`:
+	- `grant_id`
+	- `email`
+	- `ip`
+	- `timestamp`
```

Comment:
I have a naming question - what is a "grant" in this context?  My brain can't decide whether to parse it as a noun or a verb, or as past or present tense.  Which means I probably don't understand what you're driving at by tracking it.